src/core/image_resizer.js

```diff
@@ -268,6 +279,90 @@ class ImageResizer {
     return imgData;
   }
 
+  _rescaleImageData() {
```

Comment:
Nit: Can this be private?
```suggestion
  #rescaleImageData() {
```